Azure/azure-cli
# [Storage] Fix #23893, #24528: `az storage account show-connection-string/keys renew`: Fix resource group auto completion

**Related command**

`az storage account show-connection-string`
`az storage account keys renew`

**Description**

Fix #23893, #24528 

- In #24266,  we added a command validator [`validate_key_name`](https://github.com/Azure/azure-cli/blob/1a06cb7e9b4cf0a66625e2e1d958a097924fd9ce/src/azure-cli/azure/cli/command_modules/storage/commands.py#L118) for `az storage account show-connection-string`.
- `az storage account keys renew` already has command validator [`validate_key_name`](https://github.com/Azure/azure-cli/blob/1a06cb7e9b4cf0a66625e2e1d958a097924fd9ce/src/azure-cli/azure/cli/command_modules/storage/commands.py#L136)

The problem is command validator will overwrite all argument validators including [`process_resource_group`](https://github.com/Azure/azure-cli/blob/1a06cb7e9b4cf0a66625e2e1d958a097924fd9ce/src/azure-cli/azure/cli/command_modules/storage/_params.py#L290) for both commands

As a result, there's no resource group auto completion if customer provide `--account-name` without `--resource-group-name`, which will cause command failure with
```
No value for given attribute
Traceback (most recent call last):
  File "/opt/az/lib/python3.10/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/storage/operations/account.py", line 293, in show_storage_account_connection_string
    obj = scf.list_keys(resource_group_name, account_name, logging_enable=False)  # pylint: disable=no-member
  File "/opt/az/lib/python3.10/site-packages/azure/core/tracing/decorator.py", line 73, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/azure/mgmt/storage/v2022_05_01/operations/_storage_accounts_operations.py", line 1312, in list_keys
    request = build_list_keys_request(
  File "/opt/az/lib/python3.10/site-packages/azure/mgmt/storage/v2022_05_01/operations/_storage_accounts_operations.py", line 319, in build_list_keys_request
    "resourceGroupName": _SERIALIZER.url("resource_group_name", resource_group_name, 'str', max_length=90, min_length=1, pattern=r'^[-\w\._\(\)]+$'),
  File "/opt/az/lib/python3.10/site-packages/msrest/serialization.py", line 652, in url
    output = self.serialize_data(data, data_type, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/msrest/serialization.py", line 760, in serialize_data
    raise ValueError("No value for given attribute")
ValueError: No value for given attribute
```

This PR moves `validate_key_name` from command validator to argument validator as a fix.

**Testing Guide**

`az storage account show-connection-string -n $accountName --key primary`


---

This checklist is used to make sure that common guidelines for a pull request are followed.
